pages/post/index.tsx

```diff
@@ -1,38 +1,52 @@
-import NewPostEntry from "../../src/common/components/NewPostEntry";
-import { classNames } from "../../src/common/helpers/css";
-import { useNewPostStore } from "../../src/stores/useNewPostStore";
-import React, { useEffect, useState } from "react";
+import { CastRow } from "@/common/components/CastRow";
+import { useAccountStore } from "@/stores/useAccountStore";
 import { PlusCircleIcon, TrashIcon } from "@heroicons/react/24/outline";
+import { NeynarAPIClient } from "@neynar/nodejs-sdk";
+import { CastWithInteractions } from "@neynar/nodejs-sdk/build/neynar-api/v2";
 import * as Tooltip from "@radix-ui/react-tooltip";
+import { usePathname, useSearchParams } from "next/navigation";
+import React, { useEffect, useRef, useState } from "react";
 import HotkeyTooltipWrapper from "../../src/common/components/HotkeyTooltipWrapper";
+import NewPostEntry from "../../src/common/components/NewPostEntry";
+import { classNames } from "../../src/common/helpers/css";
 import { Button } from "../../src/components/ui/button";
-import { CastRow } from "@/common/components/CastRow";
-import { NeynarAPIClient } from "@neynar/nodejs-sdk";
-import { useAccountStore } from "@/stores/useAccountStore";
-import { CastWithInteractions } from "@neynar/nodejs-sdk/build/neynar-api/v2";
+import { useNewPostStore } from "../../src/stores/useNewPostStore";
 
 export default function NewPost() {
-  const { addNewPostDraft, removePostDraft, removeAllPostDrafts } =
+  const { addNewPostDraft, removePostDraft, removeAllPostDrafts, removeEmptyDrafts } =
     useNewPostStore();
   const { drafts } = useNewPostStore();
   const [parentCasts, setParentCasts] = useState<CastWithInteractions[]>([]);
   const { accounts, selectedAccountIdx } = useAccountStore();
   const selectedAccount = accounts[selectedAccountIdx];
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const savedPathname = useRef(pathname)
 
+  // get text from query params and add to text field if it exists
   useEffect(() => {
-    if (drafts.length === 0) {
```

Comment:
if there is no searchParam with text and there are no existing drafts, it's a nice user experience if a new draft is created here